skorch/callbacks/lr_scheduler.py

```diff
@@ -66,11 +72,21 @@ def __init__(self,
                  policy='WarmRestartLR',
                  monitor='train_loss',
                  event_name="event_lr",
+                 step_every='epoch',
                  **kwargs):
         self.policy = policy
         self.monitor = monitor
         self.event_name = event_name
+        self.step_every = step_every
         vars(self).update(kwargs)
+        if policy == TorchCyclicLR or policy == "TorchCyclicLR":
+            warnings.warn(
+                "The LRScheduler callback makes a step "
+                "every epoch by default from now on. To have the cyclic lr "
```

Comment:
For deprecation, we must keep the current behavior and state that the behavior will change in a Future version:

> Starting in 0.10, the LRScheduler will make a step every epoch by default. To have the cyclic lr schedule update every batch set step_every='batch'".

I am thinking of moving this logic into `initialize`:

```py
def initialize(self):
	self.step_every_ = self.step_every
    if policy == TorchCyclicLR or policy == "TorchCyclicLR" and self.step_every=='epoch':
		self.step_every_ = 'batch'
```

And then use `self.step_every_` in the other methods.

@BenjaminBossan How do you feel about this approach to deprecating the behavior? The simpler approach would be to update in `__init__`.

Reply:
Could you clarify the advantage of having both `self.step_every` and `self.step_every_`? It looks to me like having a second variable is unnecessary.

Once we decide whether to place this code in `initialize` or `init`, I'll make the changes.

Reply:
Traditionally the sckit-learn API does not allow for changing the `__init__` parameters during construction. But since this is a callback and not an estimator, we may not need to strictly follow this API contract.

Reply:
Please try to stick with the sklearn convention here and don't change the `__init__` parameters. This means making the change during `initialize` (which is equivalent to `fit` in sklearn) and use the trailing underscore. The reason why we need this is because it makes it possible to use `set_params` on callbacks.

> How do you feel about this approach to deprecating the behavior?

When it comes to deprecation: I don't have a strong opinion. From my point of view, it would also be okay to change the behavior to the new behavior, raise a warning and pass a message that explains how to get back to the old behavior. But deprecation is also totally fine with me.

Regardless of how, could you please add a `# TODO: ...` comment, @guybuk, so that we don't forget to eventually change the behavior and remove the warning?

Reply:
> When it comes to deprecation: I don't have a strong opinion. From my point of view, it would also be okay to change the behavior to the new behavior, raise a warning and pass a message that explains how to get back to the old behavior. But deprecation is also totally fine with me.

In this case, I am +0.5 with moving fast and go with the behavior change.

Let's change the behavior with a FutureWarning, we can remove the warning in `0.10`. With the behavior change, we will not need to have a `self.step_every_`.

Reply:
Done. Let me know if there's anything else.

Also, thanks for holding my hand throughout the process. I hope it was worth you time :smiley: